[CDAP-14291] Add setup for storybook for CDAP UI

ajainarayanan opened the pull request:
- Adds basic setup for storybook integration with CDAP UI
- Adds some sample storybook for references
- This page is to be used by both UI and design team as a reference for all common set of components that are being used in UI. 
- This is to ensure we have consistency of using components across UI

**Note:**
- This is an ongoing effort. As we introduce more components we could add stories with it to keep this page updated.

JIRA: https://issues.cask.co/browse/CDAP-14291
Build: https://builds.cask.co/browse/CDAP-UDUT96

tonybach commented on cdap-ui/.storybook/Welcome.tsx:
```diff
@@ -0,0 +1,50 @@
+      <p>
+        Repository of components used in CDAP UI
+      </p>
+      <img src="https://i2.wp.com/blog.codepen.io/wp-content/uploads/2018/07/happy-michael-scott.gif" />
```
lol

tonybach commented on cdap-ui/package.json:
```diff
@@ -36,19 +38,31 @@
     "@babel/preset-typescript": "^7.0.0-beta.53",
+    "@storybook/addon-actions": "^3.4.10",
+    "@storybook/addon-info": "^3.4.10",
+    "@storybook/addon-links": "^3.4.10",
```
I don't see this being used yet. Is it installed by default when you install Storybook?

tonybach commented on cdap-ui/webpack.config.common.js:
```diff
@@ -154,13 +154,6 @@ if (mode === 'production') {
-      fileName: COMMON_LIB_NAME + '.js',
-      minChunks: Infinity
-    }
-  },
```
Are these removals related to storybook?

tonybach commented on cdap-ui/.storybook/config.js:
```diff
@@ -0,0 +1,32 @@
+require('../app/cdap/styles/bootstrap_4_patch.scss');
+
+// automatically import all files ending in *.stories.js
+const req = require.context('../app/cdap/components/', true, /.stories.tsx$/);
```
Right now we only have story files ending in `.stories.tsx`, so this is fine, but should we look for `.stories.js` as well?

ajainarayanan replied:
Going forward if we write code in `ts` it would be helpful for us to have that static type check. Is there any reason you see why we would want to write stories in `js`?

tonybach reviewed (commented):
Minor comments and questions.

ajainarayanan committed:
Adds @storybook/react and associated npm modules

  - Updates .babelrc plugins to use the right plugins for object spread and class properties
  - Modifies webpack config to include parsing stories in addition to components
  - Adds a story for Alert + adds style for info alert
  - Adds story to BtnWithLoading component + Fixes BtnWithLoading component (prop types)
  - Adds story for SortableStickyGrid component
  - Modifies bootstrap_4_patch scss file to export mixin to be reused in storybook context
  - Adds types for @storybook/react
  - Adds pom build step to build storybook

ajainarayanan commented:
@elfenheart @tonybach Have added changes to cdap-ui/pom file to build storybook when run from bamboo. 

Once the PR is merged will add another build task for cdap ui develop build that will include building the storybook.

elfenheart reviewed (dismissed):
👍 LGTM

ajainarayanan committed:
Modifies node and yarn version in mvn after adding @storybook

tonybach reviewed (approved):
LGTM 👍

ajainarayanan commented:
Latest green build: https://builds.cask.co/browse/CDAP-UDUT139-2
